### PR TITLE
Migrate service-status from argparse to click

### LINKED ISSUE #39: standardize on click for cli interfaces
scripts currently use either click or argparse, because scripts from socorro-release were expected to only rely on stdlib so they could be run without installing dependencies. obs-common doesn't support such a case, so argparse scripts should be migrated to click for consistency, and also so that all scripts can be re-used in other click scripts in other obs repos if we want.

### PR DESCRIPTION
**Pull request by relud**
fixes #39

manually tested by running:
```sh
make shell
service-status # expect error message about no hosts
service-status --host=stage=https://crash-stats.allizom.org
```

I didn't add CI tests because CI shouldn't depend on github being up, but also mocking out github and an http server seemed excessive. If adding a test like that is desirable lmk.

### CODE REVIEW

File: obs_common/service_status.py [hunk 1]
```diff
@@ -9,63 +9,22 @@
...
-import argparse
 import json
-import os
 import sys
 from urllib.parse import urlparse
 from urllib.request import urlopen
+from pathlib import Path
 
+import click
+import tomllib
```
Comment (willkg) [inline obs_common/service_status.py:19]: I typically run this script from my local machine--not in a container. This changes the script to need to be run inside the container.

We might want to add a `make servicestatus` rule to services to reduce the number of steps it takes to do something we'll do semi-frequently especially during support rotations when we're trying to figure out whether to do a deploy.

File: obs_common/service_status.py [hunk 2]
```diff
@@ -141,44 +100,40 @@ def print_delta(self, main_branch, user, repo, sha):
...
+    ),
+)
+def main(main_branch, hosts):
+    config_data = {}
+    if (pyproject_toml := Path("pyproject.toml")).exists():
+        data = tomllib.loads(pyproject_toml.read_text())
+        config_data = data.get("tool", {}).get("service-status", {})
+
+    if main_branch is None:
+        main_branch = config_data.get("main_branch", "main")
```
Comment (willkg) [inline obs_common/service_status.py:127]: I tend to use `or` for things like this especially where a falsey empty string doesn't make sense:

```suggestion
    main_branch = main_branch or config_data.get("main_branch", "main")
```

**Review by willkg — APPROVED** [review]
This is a nice-to-have script and if it was broken, it's easily worked around. I don't think it needs extensive tests that increase our maintenance burden.

Looks good!

File: obs_common/service_status.py (hunk 1 above)
Comment (smarnach) [reply obs_common/service_status.py:19]: I remember vaguely that we talked about pushing obs-common to our private package index so we can `pipx install` it. Is this still the plan? If so, external dependencies should be fine.

**Review by smarnach — APPROVED** [review]

File: obs_common/service_status.py (hunk 1 above)
Comment (relud) [reply obs_common/service_status.py:19]: +1 for make servicestatus

File: obs_common/service_status.py (hunk 2 above)
Comment (relud) [reply obs_common/service_status.py:127]: I would agree, but in this case the falsey string would have to be explicitly passed as a cli flag, and I don't think we should ignore that

Comment (relud) [reply obs_common/service_status.py:127]: i slept on it an changed me mind

**Pull request merged**